### PR TITLE
refactor(@DataTable): reducers api and renaming

### PR DESCRIPTION
**Pull request by allison-c**
### Purpose 🚀
- `plugins` -> `reducers`
- Moving `<Sort />` into `<Navigation>`

### Notes ✏️
_details of code change / secondary purposes of this PR_

### Updates 📦
- [ ] MAJOR (breaking) change to _these packages_
- [ ] MINOR (backward compatible) change to _these packages_ 
- [ ] PATCH (big fix) change to _these packages_

### Storybook 📕
http://storybooks.highbond-s3.com/paprika/your-branch-name

### Screenshots 📸
_optional but highly recommended_

### References 🔗
_relevant Jira ticket / GitHub issues_

### CODE REVIEW

File: packages/DataTable/src/components/Navigation/Navigation.js [hunk 1]
```diff
@@ -4,12 +4,34 @@ import * as styled from "./Navigation.styles";
...
+  ColumnsDefinition: [],
 };
 
 export default function Navigation(props) {
-  const { children } = props;
-  return <styled.Navigation>{children}</styled.Navigation>;
+  const { children, ColumnsDefinition } = props;
+  return (
+    <styled.Navigation>
+      {React.Children.map(children, child => {
```
Comment (nahumzs) [inline packages/DataTable/src/components/Navigation/Navigation.js]: I think would be more useful to pass all the props to all component no only to the Sort.
```
<styled.Navigation>
  {React.Children.map(Children, child => React.CloneElement(child, {...props,  ...child.props)) 
</styled.Navigation>
```

File: packages/DataTable/src/components/Navigation/components/Sort/Sort.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        </DropdownMenu.Trigger>
+      )}
+    >
+      {ColumnsDefinition.map(({ props: columnProp }) => {
+        const { id: columnId, header, sortDirections } = columnProp;
+
+        if (!sortDirections || sortDirections.length === 0) return null;
+
+        return (
+          <DropdownMenu.Item key={columnId} onClick={noop}>
```
Comment (nahumzs) [inline packages/DataTable/src/components/Navigation/components/Sort/Sort.js]: why do we need to have our own button? and our own onClick, didn't that's why we have `DropDown.Item` ?

File: packages/DataTable/src/components/Navigation/Navigation.js (hunk 1 above)
Comment (allison-c) [reply packages/DataTable/src/components/Navigation/Navigation.js]: Ok! I was unsure about what kind of child components in the future, but it makes sense to keep it simple and clear

File: packages/DataTable/src/components/RowHeight/RowHeight.js [hunk 3]
```diff
@@ -61,6 +61,7 @@ export default function RowHeight(props) {
 
 RowHeight.propTypes = propTypes;
 RowHeight.defaultProps = defaultProps;
+RowHeight.displayName = "DataTable.Navigation.RowHeight";
```
Comment (nahumzs) [inline packages/DataTable/src/components/RowHeight/RowHeight.js]: Why you changed the name? This is rendered on the Navigation but the logic doesn't belong to the Navigation, also if you change this here you should change it on the `DataTable.js` where use it to extract the children with extract children.

File: packages/DataTable/src/components/Navigation/components/Sort/Sort.js (hunk 2 above)
Comment (allison-c) [reply packages/DataTable/src/components/Navigation/components/Sort/Sort.js]: From my understanding, `DropDown.Item` is each item in the dropdown list which is the blue box here:
image
But the `Item` itself is not clickable, the small pink button on the right is clickable

File: packages/DataTable/src/components/RowHeight/RowHeight.js (hunk 3 above)
Comment (allison-c) [reply packages/DataTable/src/components/RowHeight/RowHeight.js]: Maybe `DataTableControls.RowHeight` ? or do you prefer `DataTable.RowHeight` ? I understand `RowHeight` means the row height of the table, but this component is for changing the row height, the table itself doesn't care the value of row height. Actually if i remove this part https://github.com/acl-services/paprika/pull/300/files#diff-ccd22f487246440a452bd36678a0f647R23 I don't really need to know the display name of the component

File: packages/DataTable/src/components/Navigation/components/Sort/Sort.js (hunk 2 above)
Comment (nahumzs) [reply packages/DataTable/src/components/Navigation/components/Sort/Sort.js]: then I guess we don't need a menu, we can refactor it later when we are more sure about what UI we are going to have at the end. :)

**Comment by nahumzs** [comment]
lgtm :D

**Pull request merged**